lib/catalog/owid/catalog/tables.py

```diff
@@ -1255,6 +1258,29 @@ def get_unique_sources_from_tables(tables: List[Table]) -> List[Source]:
     return unique_sources
 
 
+def get_unique_origins_from_tables(tables: List[Table]) -> List[Origin]:
+    # First, ensure only "origins" (and not "sources") are used in the variables.
+    if any(
+        [
+            ("sources" in table[column].metadata) and (len(table[column].metadata.sources) > 0)
+            for table in tables
+            for column in table.columns
+        ]
+    ):
+        log.warning("Origins and sources mixed in tables. The resulting dataset may not include all origins.")
+
+    # Make a list of all origins of all variables in all tables.
+    origins = sum([table._fields[column].origins for table in tables for column in list(table.all_columns)], [])
+
+    # Get unique array of tuples of origin fields (respecting the order).
+    unique_origins_array = pd.unique([tuple(origin.to_dict().items()) for origin in origins])
```

Comment:
I'll add `__hash__` method to `Origin` so that we can use `set(...)` on it instead of doing it with dicts (your approach definitely works too, but `set` would be easier to use).

I'll do that in a different PR.